### PR TITLE
Allow building of aarch64 versions of plug-ins

### PR DESCRIPTION
**Pull request by liamellexus**
- [x] Development complete
- [ ] Documentation complete
- [ ] Test case added
- [x] Version number updated
- [ ] Representative testing complete
- [ ] Full testing complete

The plug-ins have yet to be tested and the mysql and postgres ones are not built on arm.

### CODE REVIEW

File: output/common.mk [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	LDLIBS += -lkrb5support -lkeyutils
+else
+	LDLIBS += \
+		$(shell /usr/aarch64-linux-gnu/bin/curl-config --static-libs)
+	CFLAGS += -I /usr/aarch64-linux-gnu/include
+endif
+endif
+
+TARGETS = \
```
Comment (nick-ellexus) [inline output/common.mk:27]: This line-breaking style is a bit weird.

File: Makefile [hunk 2]
```diff
@@ -48,22 +48,35 @@ package-mistral_elasticsearch:
...
 	$(MAKE) -C output/mistral_graphite package
-	$(MAKE) -C output/mistral_mysql package
 	$(MAKE) -C output/mistral_influxdb package
+	$(MAKE) -C output/mistral_mysql package
 	$(MAKE) -C output/mistral_postgresql package
 	$(MAKE) -C output/mistral_rtm package
 	$(MAKE) -C output/mistral_splunk package
+
+.PHONY: package-aarch64
+package-aarch64:
```
Comment (nick-ellexus) [inline Makefile:60]: Shouldn't the sub-makes here specify the architecture variable(s), like TGT_ARCH=aarch64 or something?

File: output/common.mk [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+$(eval $(call check_arch,$(HOST_ARCH)))
+
+# Setting TGT_ARCH before this point should allow cross-compiling
+TGT_ARCH := $(HOST_ARCH)
+
+ifeq (1, $(INCLUDE_CURL))
+# For the aach64 cross-compiler the curl-config file isn't on the path, and
+# libssl is provided by libressl which doesn't require krb5support or keyutils.
+# Also specify the include location for the header files.
+ifeq (,$(findstring aarch64,$(TGT_ARCH)))
```
Comment (nick-ellexus) [inline output/common.mk:15]: Why using findstring here and not just ifeq(aarch64,$(TGT_ARCH)) ?

File: output/common.mk [hunk 4]
```diff
@@ -0,0 +1,124 @@
+# ------------------------------------------------------------------------------
```
Comment (nick-ellexus) [inline output/common.mk:1]: Hurrah for common.mk! Add an intro comment here explaining what it's for?

File: output/mistral_fluentbit/Makefile [hunk 5]
```diff
@@ -1,113 +1,12 @@
...
-
-PACKAGE = \
-	$(PLUGIN_NAME).$(HOST_ARCH)
-
-PLUGIN_FRAMEWORK_DIR = \
-	../../common
-
-STANDARD_OBJECTS = \
-	$(PLUGIN_FRAMEWORK_DIR)/plugin_control.o
+INCLUDE_CURL = 0
```
Comment (nick-ellexus) [inline output/mistral_fluentbit/Makefile:2]: Could omit this.

File: VERSION [hunk 6]
```diff
@@ -0,0 +1 @@
+v6.2.0
```
Comment (nick-ellexus) [inline VERSION:1]: Ooh, shared VERSION?

**Review by nick-ellexus — APPROVED** [review]
Clearly a massive improvement. A few niggles.

File: VERSION (hunk 6 above)
Comment (liamellexus) [reply VERSION:1]: Yes - I got annoyed with updating each one.

File: Makefile (hunk 2 above)
Comment (liamellexus) [reply Makefile:60]: Should I change it as follows for each line?
```
$(MAKE) TGT_ARCH=aarch64 -C output/mistral_elasticsearch package
```
Or can I set ``TGT_ARCH`` once? One day I will actually understand enough of make not to have to ask this.

I guess specifying GCC=aarch64-linux-gnu-gcc here might make sense too.

**Review by nick-ellexus — APPROVED** [review]
Yaas queeeeen

**Pull request merged**